src/app/theme-chooser/themes.scss

```diff
@@ -0,0 +1,380 @@
+$light-theme-palette: igx-palette($primary: $light-color, $secondary: $royalblue-color);
+$dark-theme-palette: igx-palette($primary: $dark-color, $secondary: $orange-color);
+$black-theme-palette: igx-palette($primary: $black-color, $secondary: $yellow-color);
+
+// Define different variations of the primary color from the ligh-theme-palette.
+$primary-light-100: igx-color($light-theme-palette, "primary", 100);
+$primary-light-200: igx-color($light-theme-palette, "primary", 200);
+$primary-light-500: igx-color($light-theme-palette, "primary", 500);
+$primary-light-600: igx-color($light-theme-palette, "primary", 600);
+$primary-light-700: igx-color($light-theme-palette, "primary", 700);
+
+// Define different variations of the secondary color from the ligh-theme-palette.
+$secondary-light-100: igx-color($light-theme-palette, "secondary", 100);
+$secondary-light-200: igx-color($light-theme-palette, "secondary", 200);
+$secondary-light-500: igx-color($light-theme-palette, "secondary", 500);
+$secondary-light-600: igx-color($light-theme-palette, "secondary", 600);
+$secondary-light-700: igx-color($light-theme-palette, "secondary", 700);
+
+// Define different variations of the primary color from the dark-theme-palette.
+$primary-dark-100: igx-color($dark-theme-palette, "primary", 100);
+$primary-dark-500: igx-color($dark-theme-palette, "primary", 500);
+$primary-dark-200: igx-color($dark-theme-palette, "primary", 200);
+$primary-dark-500: igx-color($dark-theme-palette, "primary", 500);
+$primary-dark-600: igx-color($dark-theme-palette, "primary", 600);
+$primary-dark-700: igx-color($dark-theme-palette, "primary", 700);
+
+// Define different variations of the secondary color from the dark-theme-palette.
+$secondary-dark-100: igx-color($dark-theme-palette, "secondary", 100);
+$secondary-dark-200: igx-color($dark-theme-palette, "secondary", 200);
+$secondary-dark-500: igx-color($dark-theme-palette, "secondary", 500);
+$secondary-dark-600: igx-color($dark-theme-palette, "secondary", 600);
+$secondary-dark-700: igx-color($dark-theme-palette, "secondary", 700);
+
+// Define different variations of the primary color from the black-theme-palette..
+$primary-black-100: igx-color($black-theme-palette, "primary", 100);
+$primary-black-200: igx-color($black-theme-palette, "primary", 200);
+$primary-black-500: igx-color($black-theme-palette, "primary", 500);
+$primary-black-600: igx-color($black-theme-palette, "primary", 600);
+$primary-black-700: igx-color($black-theme-palette, "primary", 700);
+
+// Define different variations of the secondary color from the black-theme-palette.
+$secondary-black-100: igx-color($black-theme-palette, "secondary", 100);
+$secondary-black-200: igx-color($black-theme-palette, "secondary", 200);
+$secondary-black-500: igx-color($black-theme-palette, "secondary", 500);
+$secondary-black-600: igx-color($black-theme-palette, "secondary", 600);
+$secondary-black-700: igx-color($black-theme-palette, "secondary", 700);
```

Comment:
To me declaring the colors this way feels unnecessary and wordy. Can you reference them in the individual component themes directly, instead of creating variable references to those colors?

For instance:

```scss
$light-snackbar: igx-snackbar-theme(
  $palette: $light-theme-palette,
  $background: igx-color($light-theme-palette, 'secondary', 500),
  $text-color: $primary-light-500,
  $button-color: $primary-light-500
);
```